dali/base/dadfs.cpp

```diff
@@ -6455,11 +6458,10 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
             {
                 cost_type c;
                 if (!subfiles.item(i).getWriteCost(c,calculateIfMissing))
```

Comment:
this looks wrong - adds cost only if returns false
This is the kind of thing that we should be spotting via the regression suite. There was a JIRA related to working on that opened a while back: https://hpccsystems.atlassian.net/issues/HPCC-33208.
I've opened a sub-task, to add a single test as a POC.